New mtls connection

AryJpl opened the pull request:
New functionality supports both TLS & mTLS to mariaDB.
In order to test:

build project with "cmake -DDEBUG=ON -DMYSQL=ON . "
make clean all build
You mariaDB must be pre-configured for encrypted connections.
a) For TLS, run (Note this will fail without a proper password for usertest1)
./build/bin/ut_mysql_tls_connection
b) For mTLS, run (No password is required)
./build/bin/ut_mysql_m_tls_connection

Commit:
Related directly to the TLS  connection implementation:
1) util/src_util/ut_mysql_tls_connection.c - the unitest for a TLS connection. Note password is needed.
2) util/src_util/ut_mysql_m_tls_connection.c - the unitest for a mTLS connection. Note password is MOT needed.
3) src/src_mysql/sadb_routine_mariadb.template.c - midified sadb_init() to support TLS,mTLS, and regular non-encrypted connection.
4) crypto.h - modified Crypto_Config_MariaDB() to support extra parameters related to TLS and/or mTLS encrypted connection.
5) crypto_config.c - modified
6) crypto_config_structs.h - modified SadbMariaDBConfig_t to support parameters related to tLS/mTLS

Commit:
Related directly to the TLS  connection implementation:
1) util/src_util/ut_mysql_tls_connection.c - the unitest for a TLS connection. Note password is needed.
2) util/src_util/ut_mysql_m_tls_connection.c - the unitest for a mTLS connection. Note password is MOT needed.
3) src/src_mysql/sadb_routine_mariadb.template.c - midified sadb_init() to support TLS,mTLS, and regular non-encrypted connection.
4) crypto.h - modified Crypto_Config_MariaDB() to support extra parameters related to TLS and/or mTLS encrypted connection.
5) crypto_config.c - modified
6) crypto_config_structs.h - modified SadbMariaDBConfig_t to support parameters related to tLS/mTLS

IbraheemYSaleh reviewed (changes requested):
The only difference between:

        if (sadb_mariadb_config->encrypted_connection == 1) {
&&
        if (sadb_mariadb_config->encrypted_connection == 2) {

is that you pass in NULL or a password for the mysql_password field. They are both mTLS.

Remove the second if block and rely on the configuration to be set or default to NULL for mysql_password.

Also, you don’t need a separate unit test file for each test… just combine them into one.

Commit:
Related directly to the TLS  connection implementation:
1) util/src_util/ut_mysql_tls_connection.c - removed
2) util/src_util/ut_mysql_m_tls_connection.c - combined TLS with mTLS test
3) src/src_mysql/sadb_routine_mariadb.template.c - removed extra if statement

AryJpl commented:
@IbraheemYSaleh, The 2 changes you requested are committed, please do a pull. If you cant see them I can create a new pull request on my side. Tests are successful: 
![image](https://user-images.githubusercontent.com/94404852/148416971-6004b4b0-c846-4f0a-aa5b-7b77e7452049.png)
****

Commit:
Related directly to the TLS  connection implementation:
1) util/src_util/ut_mysql_m_tls_connection.c - removed password
